lib.rs

```diff
@@ -1000,4 +1004,43 @@ pub mod tests {
         assert_eq!(vec.clone().into_iter().len(), 3);
         assert_eq!(vec.drain().len(), 3);
     }
+
+    #[bench]
+    fn bench_push(b: &mut Bencher) {
+        b.iter(|| {
+            let mut vec: SmallVec<[u64; 16]> = SmallVec::new();
+            for x in 0..100 {
```

Comment:
I believe if the closure returns a value then the optimizer will not remove it.

I was empirically able to determine that the optimizer was not removing the code either way (for now), but I will include the vec return value to ensure it.